### DISCUSSION TITLE
bsconfig.json

### DISCUSSION THREAD

**Discussion by E5TE** [rokucommunity/vscode-brightscript-language#382] [General]
Hi @TwitchBronBron ,
I just started with Roku's BrightScript language and installed the plug-in in the vs-code.
Since I am still learning the most important feature to me for now is to jump from references to definitions, and basic code navigations.
This feature will work once I create and config the file 'bsconfig'? and if it, can you please provider a sample of the 'bsconfig' file in order to understand how to do that?
Thanks in advance

**Comment by TwitchBronBron** [comment]
Hey, @E5TE!

The vscode extension works out of the box with the standard Roku project structure (`source/`, `components/`, etc). However, if you have additional folders, or your project resides entirely in a subdirectory, that's when you would need to create a bsconfig.json file. 

I realized that our documentation was a bit lacking, so I just updated them. Take a look at the [docs](https://rokucommunity.github.io/vscode-brightscript-language/Editing/index.html) to see some examples of different project structures. 

Feel free to reach to us on [Slack](https://join.slack.com/t/rokudevelopers/shared_invite/zt-4vw7rg6v-NH46oY7hTktpRIBM_zGvwA) as well, or we can continue to chat here in Discussions.

**Comment by E5TE** [comment]
@TwitchBronBron , thanks for your quick reply and action :)
I did it as you described and works perfect!
Thanks again
